server/commands/teamCreator.js

```diff
@@ -6,6 +6,12 @@ import { generateAvatarUrl } from "../utils/avatars";
 
 const log = debug("server");
 
+type TeamCreatorResult = {|
+  team: Team,
+  authenticationProvider: AuthenticationProvider,
+  isNew: boolean,
```

Comment:
`isNewTeam` is fine here, no need to be quite so general. That ways you can use `isNewTeam` and `isNew User` and the commands can be symmetrical and completely semantically coherent